core/cache/application_test.go

```diff
@@ -23,10 +23,17 @@ var _ = gc.Suite(&ApplicationSuite{})
 func (s *ApplicationSuite) TestConfigIncrementsReadCount(c *gc.C) {
 	m := s.NewApplication(appChange, nil)
 	c.Check(testutil.ToFloat64(s.Gauges.ApplicationConfigReads), gc.Equals, float64(0))
+
 	m.Config()
 	c.Check(testutil.ToFloat64(s.Gauges.ApplicationConfigReads), gc.Equals, float64(1))
 	m.Config()
 	c.Check(testutil.ToFloat64(s.Gauges.ApplicationConfigReads), gc.Equals, float64(2))
+
+	// Goroutine safety.
+	go m.Config()
+	go m.Config()
+	go m.Config()
+	c.Check(testutil.ToFloat64(s.Gauges.ApplicationConfigReads), gc.Equals, float64(5))
```

Comment:
This looks flaky as you cannot guarantee that the go-routines will spin up and actually run before you hit the check (on a fast machine I suspect this test will fail). How about changing this to :

```go
var wg sync.WaitGroup
wg.Add(3)
for i:=0;i<3;i++{
  go func(){ 
     defer wg.Done()
     m.Config()
  }()
}
wg.Wait()
c.Check(...)
```

Reply:
Rookie mistake. Yes.